src/sys/socket/addr.rs

```diff
@@ -748,6 +798,53 @@ impl SockAddr {
                 #[cfg(any(target_os = "ios", target_os = "macos"))]
                 Some(AddressFamily::System) => Some(SockAddr::SysControl(
                     SysControlAddr(*(addr as *const sys_control::sockaddr_ctl)))),
+                // FIXME: android is same ?
+                // #[cfg(any(target_os = "android", target_os = "linux"))]
+                #[cfg(target_os = "linux")]
+                Some(AddressFamily::Packet) => {
+                    let sll: *const libc::sockaddr_ll = mem::transmute(addr);
+                    let mac = EtherAddr((*sll).sll_addr[0],
+                                      (*sll).sll_addr[1],
+                                      (*sll).sll_addr[2],
+                                      (*sll).sll_addr[3],
+                                      (*sll).sll_addr[4],
+                                      (*sll).sll_addr[5]);
+                    Some(SockAddr::Ether(mac))
+                },
+                #[cfg(any(target_os = "dragonfly",
+                  target_os = "freebsd",
```

Comment:
Please indent so `target_os` is always vertically aligned. There are other places in this document you can reference for the specific style.

Reply:
okay